heuristic/classes/Stack.py

```diff
@@ -56,6 +57,28 @@ def push_front(self, item: Item):
         self._set.add(item)
         self._volume += item.volume
 
+    def push_effective_front(self, item: Item):
+        """
+        Places the item in the front of the truck (right). If there are pickups
+        in the front already, places them behind them. # TODO O(n)?.
+        """
+        if len(self.stack) == 0:
+            self.push_front(item)
+
+            return
+        
+        if self.stack[-1].destination != DEPOT:
+            self.push_front(item)
+        else:
+            stack = list(reversed(self.stack))
+            idx = next(
```

Comment:
There is probably a better way to do this (you should almost never need to use the next(generator) idiom with these sorts of generators), but I don't know what you're attempting to do here - it certainly needs more explanation.